src/data_structure/economic_structure.jl

```diff
@@ -0,0 +1,491 @@
+                        elseif t_start < start_of_operation
+                            val = max(min(1-(start_of_operation-t_start)/dur,1),0)
+                        #=NOTE:
+                        if the end of timeslice t is after the start of operation and the start of the timeslice t
+                        is before the start of operation, val will take a value between 0 and 1, depending on
+                        how much of this capacity is available on average during t=#
+                        else
+                            val = max(min((end_of_operation-t_start)/dur,1),0)
+                        #=NOTE:
+                        in all other cases, val will describe the fraction [0,1] that is (still) available at
+                        time step t. This will be 1, if the technology does not get decomissioned during t,
+                        a fraction, if the technology gets decomssioned during t, and 0 for all other cases (fully decomissioned)
+                        =#
+                        end
+                        capacity_transfer_factor[(id, vintage_t.start.x, t.start.x)] =  parameter_value(val)
+                        push!(timeseries_val,val)
+                        push!(timeseries_ind,t_start)
+                    end
+                    push!(map_indices,vintage_t_start)
+                    push!(timeseries_array,TimeSeries(timeseries_ind,timeseries_val,false,false))
+                end
+                push!(map_stoch_indices,s)
+                push!(map_inner, SpineInterface.Map(map_indices,timeseries_array))
+            end
+            obj_cls.parameter_values[id][param_name] = parameter_value(SpineInterface.Map(map_stoch_indices,map_inner))
+            #NOTE: map_indices here will be stochastic_scenarios!
+        else
+            obj_cls.parameter_values[id][param_name] = parameter_value(1)
+        end
+    end
+    @eval begin
+        $(param_name) = $(Parameter(param_name, [obj_cls]))
+    end
+end
+
+
+
+
+"""
+    generate_conversion_to_discounted_annuities()
+
+The conversion_to_discounted_annuities factor translates the overnight costs of an investment
+into discounted (to the `dicount_year`) annual payments, distributed over the total
+lifetime of the investment. Investment payments are assumed to increase linearly over the lead-time, and decrease
+linearly towards the end of the economic lifetime.
+
+"""
+function generate_conversion_to_discounted_annuities!(m::Model, obj_cls::ObjectClass, obj_name::Symbol)
+    instance = m.ext[:spineopt].instance
+    discnt_year = discount_year(model=instance)
+    conversion_to_discounted_annuities = Dict()
+    investment_indices = eval(Symbol("$(obj_name)s_invested_available_indices"))
+    lead_time = eval(Symbol("$(obj_name)_lead_time"))
+    econ_lifetime = eval(Symbol("$(obj_name)_investment_econ_lifetime"))
+    param_name = Symbol("$(obj_name)_conversion_to_discounted_annuities")
```

Comment:
This eval and Symbol interpolation are not the most readable, would you consider passing all the arguments explicitely?

Reply:
@manuelma thank you. I tried something to explicitly define the variables/parameters (https://github.com/spine-tools/SpineOpt.jl/pull/929/commits/992f1853ad1ec8cfb797bf179eb3efc9a9316c04), do you think this seems fine? Or do you suggest something different? Once we agree on this, I'll refactor the rest.